### DISCUSSION TITLE
[2023-10-30] GUI Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8184] [Weekly Demo Day]
- Most of the last week, I was busy being sick. 
- Besides all meetings and reviews, I progressed a little with #8070 which makes aligning nodes easier when dragging them.

**Comment by kazcw** [comment]
- Merged #8073 after adding new logic for the new AST and refactoring to fit into the reorganized `GraphEditor`.
- Implemented #8157 (in CI).

Working on a new synchronization representation that will have better edit conflict resolution, and better undo/redo behavior (#8134).

**Comment by MichaelMauderer** [comment]
* Continued and finally finished work on #8183

[Peek 2023-10-30 14-44.webm](https://github.com/enso-org/enso/assets/1428930/33e96aff-fa0b-48b4-b31d-fa486fd29021)


* Started work on #8015

**Comment by vitvakatu** [comment]
- Documentation panel PR was finished and merged: https://github.com/enso-org/enso/pull/8118
- Navigation in documentation panel implemented: https://github.com/enso-org/enso/pull/8142 and https://github.com/enso-org/enso/pull/8176


https://github.com/enso-org/enso/assets/6566674/7daed14a-34bc-4996-877b-e45aa2361794

**Comment by Frizi** [comment]
- Finished integration of gui2 demo scenes, added a way to provide mock engine responses and skipping Yjs server connection.
- Working on selection and rendering of node widgets. So far have working basic numeric slider, but have many issues with unstable ID map that must be resolved. #7732

https://github.com/enso-org/enso/assets/919491/13d51c46-181c-4f64-ba8b-8930e4a6d343

**Comment by somebody1234** [comment]
- mostly website and dashboard stuff last week

- address reviews (mostly writing and refactoring tests) for #8112
- finishing up #8180 (under the `visualizations` directory)
  - reminder that GUI2 uses Vue visualizations which are incompatible with GUI1 class-based visualizations.

(the video is a bit underwhelming, due to a poor choice of example projects + visualizations to use)

https://github.com/enso-org/enso/assets/4046547/58b17727-8139-4bf0-a26b-b127f005ca1c

**Comment by mwu-tow** [comment]
Last week:
* Finishing #7863 — PR expected today.
* Observing macOS notarization
* CI: Python2 issue

**Comment by wdanilo** [comment]
Website progress:

1. The pricing page was added, bot was integrated, and logging-in works (thanks E-Hern!).
2. All sections will be finished today (I'm finishing last one – "automate & deploy").
3. There is a lot of small bugs (icons appearing in random places, bad spacing, animations firing wrongly) – this is the focus for this week.
4. Focus for this week is also correct support for all screen sizes.

You can check out the website [here](https://enso-org.github.io/website2), but do it for now full-screen using Chrome.


https://github.com/enso-org/enso/assets/1623053/cb3ceafd-ea49-42c3-85b6-4473c11e13cd

**Comment by JaroslavTulach** [comment]
Engine team:
- [inline evaluation](https://github.com/enso-org/enso/pull/8148) - e.g. _preview visualization without modifying the source code_ - in progress
- speeding up startup is in progress - #8160
- [upgrading GraalVM](https://github.com/enso-org/enso/pull/7991) is progressing well
